Chore: (Docs)  Doc Blocks updates

jonniebigodes opened the pull request:
With this pull request, the Doc Blocks documentation is updated for clarity, opening up that section of the documentation for further additions/improvements.

Closes #15653 and #14244.

What was done:
- The Docs Blocks documentation was broken down into smaller pages for clarity with examples.
- Created snippets for the changes introduced.
- Created additional assets for the documentation


Feel free to provide feedback

jonniebigodes committed:
docs changes for doc blocks

shilman commented on docs/toc.js:
```diff
@@ -116,9 +116,51 @@ module.exports = {
+          children: [
+            {
+              pathSegment: 'doc-block-argstable',
+              title: 'Argstable',
```
```suggestion
              title: 'ArgsTable',
```

jonniebigodes replied:
Fixed with the latest commit. Thanks for the callout, appreciate it 🙏 . Marking this as solved.

shilman commented on docs/writing-docs/doc-block-argstable.md:
```diff
@@ -0,0 +1,189 @@
+
+<div class="aside">
+
+💡 Currently, there's an issue when using MDX stories with IE11. This issue does <strong>not</strong> apply to [DocsPage](./docs-page.md). If you're interested in helping us fix this issue, read our
```
I don't think this belongs here

shilman commented on docs/writing-docs/doc-block-canvas.md:
```diff
@@ -0,0 +1,73 @@
+
+<div class="aside">
+
+💡 Currently, there's an issue when using MDX stories with IE11. This issue does <strong>not</strong> apply to [DocsPage](./docs-page.md). If you're interested in helping us fix this issue, read our
```
Can we put this at the top of the MDX page and not on each of the DocBlocks page? I don't think it's a common enough problem to merit all the space. WDYT?

shilman replied:
Also I don't think it's something that can be fixed with a contribution since AFAIK it's a problem with the MDX library itself, not our implementation

jonniebigodes replied:
@shilman with the latest commit, I've removed the aside from all the docs blocks pages. I'll make a follow-up pull request polishing the wording on the MDX page. Sounds good?

shilman reviewed (commented):
Looking great -- minor comments below 🙏

jonniebigodes committed:
Updates per feedback.

jonniebigodes commented:
@shilman Self merging this. I'll make the follow-up pull request, addressing your comment later today! Appreciate the time you took to look at this and the feedback 🙏
